tls/pk.c

```diff
@@ -23,18 +23,192 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-#include "debug.h"
-#include "pk.h"
-#include "pk_internal.h"
-#include "rsa.h"
 #include "tls_internal.h"
 #include "ecp.h"
-#include "ecdsa.h"
+#include "mpool.h"
+#include "pk.h"
+#include "rsa.h"
+
+/* Defined in ecdsa.c */
+extern int ttls_ecdsa_write_signature(TlsEcpKeypair *ctx,
+				      const unsigned char *hash, size_t hlen,
+				      unsigned char *sig, size_t *slen);
+extern int ttls_ecdsa_read_signature(TlsEcpKeypair *ctx,
```

Comment:
Placing prototypes here is dangerous. If a prototype is in the header file, and during some refactoring parameter list in the function definition changes, compiler will warn about the ismatch. But if the prototype only defined here, no warnings and no errors will occur, but arguments won't be passed as intended.
So, why?

Reply:
I just don't see much sense to keep a separate `.h` file just for 2 function prototypes and which is used in one place only.

Reply:
tl;dr it makes compiler check what otherwise needs to be checked manually

---

Separate header files make code a little bit more robust during future changes.
So for example, if code look like this:
```c
/* func.h */
#ifndef _header_h
#define _header_h

int func(int param1);

#endif
```
```c
/* func.c */
#include "func.h"

int func(int param1) {
  return param1 * param1;
}
```
```c
/* main.c */
#include "func.h"
int main(void) {
  return func(1);
}
```
Then change in any of the files will result in compiler error.
But if `func()` prototype is declared in `main.c`, and `func.h` doesn't exist, it's possible to change `func()` definition to
```c
int func(int param1, int param2) {
  return param1 * param1 + param2 * param2;
}
```
without realizing that `main.c` still calls `func(1)`.

Reply:
Yeah, makes sense, thank you. Let's move back to the header later if we really change the function signatures - 2 lines files look quite weird :)